.github/workflows/pr.yaml

```diff
@@ -0,0 +1,60 @@
+name: "PR CI"
+
+# Trigger when a pull request is created or code is updated as well as when the PR is merged to main
+on:
+  pull_request:
+    types: [opened, synchronize, ready_for_review]
+  push:
+    branches:
+      - main
+
+env:
+  CACHE_NAME: ci
+
+jobs:
+  build_and_test:
+    name: "Build and Test"
+    if: ${{ github.event_name == 'push' || github.event.pull_request.draft == false }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node_version: [14.x]
+    steps:
+      - name: "Setup - Checkout"
+        uses: actions/checkout@v2.3.4
+
+      - name: "Setup - Use Node.js ${{ matrix.node_version }}"
+        uses: actions/setup-node@v2.1.5
+        with:
+          node-version: ${{ matrix.node_version }}
+      
+      - name: "Cache node modules"
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-modules-${{ env.CACHE_NAME }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-modules-${{ env.CACHE_NAME }}-
```

Comment:
I'm very much not a YAML person but is the trailing dash needed here?

Reply:
It's to prevent it picking up a cache without a hash on the end... I'm not overly convinced that the cache stuff has been implemented correctly at all but I've not had chance to fully get my head round it and run any tests so for now it's just a copy of every other workflow for simplicity